### ISSUE TITLE
`KaldiRecognizer` instance reset

### ISSUE THREAD

**Issue by mhu-coder** [alphacep/vosk-api#146]
Hello,
Thanks for this amazing work!

I had a small question regarding the `KaldiRecognizer` instance: is it normal that I can't use the `AcceptWaveform` method once the `FinalResult` method has been called?  
Looking at the source code, I can see that the private `AcceptWaveform` method calls the `CleanUp` method when `input_finalized_` has been set, which is the case when `FinalResult` is called. However, in the `CleanUp` method, the `feature_pipeline` is only reset when `frame_offset > 20000`. In this case, if `FinalResult` is called before 20000 frames have been seen, I get the following `kaldi::KaldiFatalError`:

```
ERROR (VoskAPI:AcceptWaveform():online-feature.cc:136) AcceptWaveform called after InputFinished() was called.
terminate called after throwing an instance of 'kaldi::KaldiFatalError'
  what():  kaldi::KaldiFatalError
Aborted (core dumped)
```

Would it make sense to modify the condition `if (frame_offset > 20000)` to `if (input_finalized_ || frame_offset > 20000)` ? This modification would also effectively bypass the else clause given where `CleanUp` is called :thinking: .

For the background, I would like to compare the `KaldiRecognizer` against the `online2-tcp-nnet3-decode-faster`, which I have modified for a usage in Web Assembly (cf [demo](https://kaldi-web.loria.fr/) and source code [here](https://gitlab.inria.fr/kaldi.web/kaldi-wasm)). The problem with `online2-tcp-nnet3-decode-faster` is the static decoding graph as it may be too large to be loaded on devices with limited memory. So I was hoping that the `KaldiRecognizer` would solve this memory issue while introducing negligible computation overhead due to the dynamic build of the graph.

Cheers,
Mathieu

**Comment by nshmyrev** [comment]
> I had a small question regarding the KaldiRecognizer instance: is it normal that I can't use the AcceptWaveform method once the FinalResult method has been called?

KaldiRecognizer object is very lightweight, you can simply create a new one if you finalized previous one.

> For the background, I would like to compare the KaldiRecognizer against the online2-tcp-nnet3-decode-faster, which I have modified for a usage in Web Assembly (cf demo and source code here). 

Cool, you can check our demo here too: https://github.com/alphacep/vosk-api/issues/49

**Comment by nshmyrev** [comment]
But the idea to rerun the cleanup is probably good too.

**Comment by mhu-coder** [comment]
Thanks for the quick answer, much appreciated :). I guess my confusion comes from how `Result` is expected to be called, compared to `PartialResult` and `FinalResult`.

**Comment by nshmyrev** [comment]
I have implemented this idea in e5c08f7710dd5201970d930d58602b896480aae4, thank you!
